HacktoberfestProject.Web/Services/TableService.cs

```diff
@@ -18,20 +19,41 @@ public TableService(IUserRepository userRepository)
             _userRepository = userRepository ?? throw new ArgumentNullException(nameof(userRepository));
         }
 
-        public Task<ServiceResponse<IEnumerable<Pr>>> AddPrByUsernameAsync(string username, Pr pr)
+        public async Task<ServiceResponse<Pr>> AddPrByUsernameAsync(string username, string owner, string repositoryName, Pr pr)
         {
-            throw new NotImplementedException();
+            var user = await GetUser(username);
+
+            Repository repository = GetReporitory(user , owner, repositoryName);
+
+            var serviceResponse = new ServiceResponse<Pr>
+            {
+                ServiceResponseStatus = ServiceResponseStatus.Created
+            };
+
+            if (!repository.Prs.All(p => p.PrId == pr.PrId))
+            {
+                repository.Prs.Add(pr);
+                user.RepositoryPrAddedTo.Add(repository);
+                await _userRepository.UpdateAsync(user);
+                serviceResponse.Message = "Pr Added!";
+            }
+            else
+            {
+                serviceResponse.Message = "Pr was already added!";
+                serviceResponse.ServiceResponseStatus = ServiceResponseStatus.Ok;
```

Comment:
We should add another responseEnum to reflect this outcome

Reply:
understood and will implament